annotations/views/rest_views.py

```diff
@@ -493,8 +500,71 @@ def _relabel(datum):
                 'identifier': 'uri'
             }
 
-            return {_fields.get(k, k): v for k, v in datum.iteritems()}
-        return Response({'results': map(_relabel, [c.data for c in concepts])})
+            return {_fields.get(k, k): v for k, v in datum.iteritems() }
+        results = map(_relabel, [c.data for c in concepts])
+
+        def _parseViaf(self, uri, new_concepts):
+            data = urllib2.urlopen(uri)
+            tree = e.parse(data)
+            root = tree.getroot()
+            namespace = {'foaf': 'http://xmlns.com/foaf/0.1/', 'ns1': 'http://viaf.org/viaf/terms#', 'owl': 'http://www.w3.org/2002/07/owl#', 'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#', 'void': 'http://rdfs.org/ns/void#'}
+            find = e.XPath("//ns1:text", namespaces = namespace)
+            concept_name = find(root)[0].text
+            concept_description = find(root)[0].text
+
+            dic = {
+                'label': concept_name,
+                'desc': concept_description,
+                'uri': uri,
+                'auth': "VIAF"}
+            new_concepts.append(dic)
+            return new_concepts
+
+        def _parseHps(self, uri, new_concepts):
+            data = urllib2.urlopen(uri)
+            tree = ET.parse(data)
+            root = tree.getroot()
+            namespace = {'hps': 'http://www.digitalhps.org/'}
+
+            for entry in root.findall('hps:conceptEntry', namespace):
+                description1 = entry.find('hps:description', namespace)
+                name1 = entry.find('hps:lemma', namespace)
+                concept_description = description1.text
+                concept_name = name1.text
+
+            dic = {
+                'label': concept_name,
+                'desc': concept_description,
+                'uri': uri,
+                'auth': "Concept Power"}
+            new_concepts.append(dic)
+            return new_concepts
+        
+        for result in results:
+            concepts = []
```

Comment:
do you really need this array? can't you work just with result["identities"]?

Reply:
@jdamerow If I remove concepts what will I compare results["identities"] with?